openfermioncirq/gates/fermionic_simulation.py

```diff
@@ -161,12 +235,30 @@ def qubit_generator_matrix(self) -> np.ndarray:
     def fermion_generator(self) -> openfermion.FermionOperator:
         """The FermionOperator G such that the gate's unitary is exp(-i t G)
         with exponent t using the Jordan-Wigner transformation."""
-        half_generator = sum(
-            (w * G
-             for w, G in zip(self.weights, self.fermion_generator_components)),
-            openfermion.FermionOperator())
+        half_generator = sum((
+            w * G
+            for w, G in zip(self.weights, self.fermion_generator_components())),
+                             openfermion.FermionOperator())
         return half_generator + openfermion.hermitian_conjugated(half_generator)
 
+    def to_interaction_operator(
```

Comment:
This method should not be defined on ParityPreservingFermionicGate. For instance, the gate could be using higher-order or non-number-preserving terms not supported by InteractionOperator. Also, it shouldn't be called `to_interaction_operator`, because that sounds like it is simply converting from one format to another. Maybe `interaction_operator_generator` or something.

Reply:
See above.